hazelcast/src/test/java/com/hazelcast/test/backup/TestBackupUtils.java

```diff
@@ -0,0 +1,344 @@
+     */
+    public static <K, V> BackupAccessor<K, V> newMapAccessor(HazelcastInstance[] cluster, String mapName, int replicaIndex) {
+        return new MapBackupAccessor<K, V>(cluster, mapName, replicaIndex);
+    }
+
+    /**
+     * Create a new instance of {@link BackupAccessor} for a given cache.
+     * It access a first backup replica.
+     *
+     * @param cluster all instances in the cluster
+     * @param cacheName cache to access
+     * @param <K> type of keys
+     * @param <V> type of values
+     * @return accessor for a given cache and first backup replica
+     */
+    public static <K, V> BackupAccessor<K, V> newCacheAccessor(HazelcastInstance[] cluster, String cacheName) {
+        return newCacheAccessor(cluster, cacheName, DEFAULT_REPLICA_INDEX);
+    }
+
+    /**
+     * Create a new instance of {@link BackupAccessor} for a given cache.
+     * It allows to access an arbitrary replica index as long as it's greater or equals 1
+     *
+     * @param cluster all instances in the cluster
+     * @param cacheName cache to access
+     * @param replicaIndex replica index to access
+     * @param <K> type of keys
+     * @param <V> type of values
+     * @return accessor for a given cache and replica index
+     */
+    public static <K, V> BackupAccessor<K, V> newCacheAccessor(HazelcastInstance[] cluster, String cacheName, int replicaIndex) {
+        return new CacheBackupAccessor<K, V>(cluster, cacheName, replicaIndex);
+    }
+
+    public static <K, V> void assertBackupEntryEqualsEventually(final K key, final V expectedValue, final BackupAccessor<K, V> accessor) {
+        assertTrueEventually(new AssertTask() {
+            @Override
+            public void run() {
+                V actualValue = accessor.get(key);
+                assertEquals("Backup entry with key '" + key + "' was '" + actualValue + "', but it was expected to be '"
+                        + expectedValue + "'.", expectedValue, actualValue);
+            }
+        });
+    }
+
+    public static <K, V> void assertBackupEntryNullEventually(final K key, final BackupAccessor<K, V> accessor) {
+        assertTrueEventually(new AssertTask() {
+            @Override
+            public void run() {
+                V actualValue = accessor.get(key);
+                assertNull("Backup entry with key '" + key + "' was '" + actualValue + "', but it was expected to be NULL.", actualValue);
+            }
+        });
+    }
+
+    public static <K, V> void assertBackupSizeEventually(final int expectedSize, final BackupAccessor<K, V> accessor) {
+        assertTrueEventually(new AssertTask() {
+            @Override
+            public void run() {
+                int actualSize = accessor.size();
+                assertEquals("Expected size: ' " + expectedSize + "' actual size: " + actualSize, expectedSize, actualSize);
+            }
+        });
+    }
+
+
+    // ######################################################################
```

Comment:
implementations of accessors is a pretty awful & fragile code to be honest. but it's isolated awfulness :)